modules/javafx.controls/src/main/java/javafx/scene/control/skin/TableColumnHeader.java

```diff
@@ -660,11 +660,11 @@ private <T,S> void resizeColumnToFitContent(TableView<T> tv, TableColumn<T, S> t
 
         int rows = maxRows == -1 ? items.size() : Math.min(items.size(), maxRows);
         double maxWidth = 0;
+        cell.updateTableColumn(tc);
+        cell.updateTableView(tv);
         for (int row = 0; row < rows; row++) {
             tableRow.updateIndex(row);
 
-            cell.updateTableColumn(tc);
-            cell.updateTableView(tv);
             cell.updateTableRow(tableRow);
```

Comment:
> can this also be moved out? `cell.updateTableRow(tableRow);`

I don't think so. Since tableRow gets an updated index above, the cell's style could change (e.g. odd / even styling).

Reply:
Well, the `tableRow` is still set (but before the loop), we just update the index in this loop. 
It looks like setting the table row in this loop is not needed, just once above. I also don't see that `setTableRow` has any side effects. So maybe worth a try.

Reply:
You're right. I tried it out and it seems to work.

Reply:
thank you for the measurements!